python_frank_energie/frank_energie.py

```
@@ -19,251 +45,999 @@ class FrankEnergie:
+                __typename
+            }
+            """,
+            "MonthSummary",
+            {"siteReference": site_reference},
+        )
+
+        try:
+            response = await self._query(query)
+            return MonthSummary.from_dict(response)
+        except Exception as e:
+            raise FrankEnergieException(f"Failed to fetch month summary: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)